src/components/tree-item/tree-item.tsx

```diff
@@ -361,7 +365,9 @@ export class TreeItem
 
   @Listen("click")
   onClick(event: Event): void {
-    if (this.disabled) {
+    const composedPath = event.composedPath();
+    const isActionClick = composedPath.includes(this.actionSlotWrapper);
+    if (this.disabled || isActionClick) {
```

Comment:
Would we need to add in support for keyboard users here for consistency? cc @driskull 

### keyboard
![keyboard-tree](https://user-images.githubusercontent.com/5023024/236551522-1759648c-f81a-41de-8559-39e2a96dd5b1.gif)

### mouse
![mouse-tree](https://user-images.githubusercontent.com/5023024/236551664-82f65e4e-6753-46ef-b9d2-d06d924935aa.gif)

Reply:
oh sure I can handle that case

Reply:
A11y wise, if there are slotted actions the role may need to change from tree to treegrid. Not sure though.

Reply:
Yeah there are some other accessibility updates that would be good to make on the tree component as a whole. I could probably try to dig into that next release as it's part of our roadmap for Online